packages/python/plotly/plotly/express/_imshow.py

```diff
@@ -137,11 +153,33 @@ def imshow(
 
     In order to update and customize the returned figure, use
     `go.Figure.update_traces` or `go.Figure.update_layout`.
+
+    If an xarray is passed, dimensions names and coordinates are used for
+    axes labels and ticks.
     """
     args = locals()
     apply_default_cascade(args)
+    img_is_xarray = False
+    if xarray_imported:
+        if isinstance(img, xarray.DataArray):
+            y_label, x_label = img.dims[0], img.dims[1]
```

Comment:
should we include units like they do in xarray?

Reply:
I decided not to because the logics start to be complicated, especially with `labels` (should we add the units also when the label was overriden with `labels`?).

Reply:
I think we should have the default label include the units, but then let the user override the whole thing. Otherwise we'll get users saying "but the built-in matplotlib version does all this stuff"

Reply:
ok, I was reluctant to code too much xarray-specific stuff but then I figured out that we could just use their function for this :-).